docs/schema-manager.md

```diff
@@ -0,0 +1,85 @@
+# FHIR Schema Manager
```

Comment:
This doc is very useful :) Should we somehow provide link to this doc on [sql schema migration guide](https://github.com/microsoft/fhir-server/blob/main/docs/SchemaMigrationGuide.md)